### PR TITLE
Enhancement/2968/add summary card

### PR DESCRIPTION
**Pull request by SriHV**
### What is the context of this PR?

Added a new summary variant where the groups are rendered in a card layout as shown in this [prototype](https://eq-runner-prototypes.netlify.app/prototypes/summary-card/)
### How to review this PR
Navigate to the summary components and compare the new  `example-summary-card-grouped` with the available prototype

### Checklist

This needs to be completed by the person raising the PR.

- [x] I have selected the correct Assignee
- [x] I have linked the correct Issue

### CODE REVIEW

File: src/components/summary/_macro.njk [hunk 1]
```diff
@@ -9,24 +9,27 @@
...
                     {% endif %}
                     {% if group.rows %}
-                        <div class="ons-summary__items">
+                        <div class="{{"ons-summary__items--card" if params.variant == 'card' and not params.hub else "ons-summary__items"}}">
 
                             {% for row in group.rows %}
                                 {% set itemClass = "" %}
                                 {% if row.error %} {% set itemClass = " ons-summary__item--error" %}{% endif %}
                                 {% if row.total %} {% set itemClass = itemClass + " ons-summary__item--total" %}{% endif %}
+                                {%if params.variant == "card" %} {% set itemClass = "--card" %} {% endif %}
```
Comment (rmccar) [inline src/components/summary/_macro.njk:32]: This logic wont work if `row.error` or `row.total` are set alongside `params.variant` being set to "card"

File: src/components/summary/_macro.njk [hunk 2]
```diff
@@ -9,24 +9,27 @@
...
+
     <div class="{{ className }}">
         {% for summary in params.summaries %}
             {% if summary.summaryTitle %}
                 <h2 class="ons-summary__title ons-u-mb-m{{ " ons-u-mt-m" if loop.index > 1 else "" }}">{{ summary.summaryTitle }}</h2>
                 {% set titleSize = "3" %}
             {% endif %}
             {% for group in summary.groups %}
-                <div {% if group.id %}id="{{ group.id }}" {% endif %} class="ons-summary__group">
+                <div {% if group.id %}id="{{ group.id }}" {% endif %} class="{{"ons-summary__group--card" if params.variant == 'card' and not params.hub else "ons-summary__group"}}">
```
Comment (rmccar) [inline src/components/summary/_macro.njk:21]: This logic can be simplified and repetition removed if you just set a var at the beginning and then add it to the end of these classes. Adding nothing if the condition isn't met

File: src/components/summary/_summary.scss [hunk 3]
```diff
@@ -46,6 +51,12 @@ $hub-row-spacing: 1.3rem;
       background: var(--ons-color-errors-tint);
       border-left: 8px solid var(--ons-color-errors);
     }
+
+    &--card {
+      &:not(:last-child){
+        border-bottom: 1px solid var(--ons-color-borders);
```
Comment (rmccar) [inline src/components/summary/_summary.scss:57]: Colour also needs changing here
```suggestion
        border-bottom: 1px solid var(--ons-color-borders-list);
```

**Review by rmccar — COMMENTED** [review]
Some things to look at. I actually think we might be able to rename the `--ons-color-borders-list` variable to something like `--ons-color-borders-light` considering if we use it for this then it will no longer just be used for lists

**Comment by rmccar** [comment]
We will probably need to cover some of this new functionality with some tests too

File: src/components/summary/_macro.njk [hunk 4]
```diff
@@ -9,26 +11,27 @@
...
+
     <div class="{{ className }}">
         {% for summary in params.summaries %}
             {% if summary.summaryTitle %}
                 <h2 class="ons-summary__title ons-u-mb-m{{ " ons-u-mt-m" if loop.index > 1 else "" }}">{{ summary.summaryTitle }}</h2>
                 {% set titleSize = "3" %}
             {% endif %}
             {% for group in summary.groups %}
-                <div {% if group.id %}id="{{ group.id }}" {% endif %} class="ons-summary__group">
+                <div {% if group.id %}id="{{ group.id }}" {% endif %} class="ons-summary__group {{"ons-summary__group--card" if cardVariant}}">
```
Comment (rmccar) [inline src/components/summary/_macro.njk:22]:
```suggestion
                <div {% if group.id %}id="{{ group.id }}" {% endif %} class="ons-summary__group{{ variant }}">
```
Along with this

File: src/components/summary/_macro.njk [hunk 5]
```diff
@@ -1,6 +1,8 @@
 {% macro onsSummary(params) %}
     {% set className = "ons-summary" %}
     {% set titleSize = "2" %}
+    {% set cardVariant =  params.variant if params.variant == "card" and not params.hub else ""%}
```
Comment (rmccar) [inline src/components/summary/_macro.njk:4]:
```suggestion
    {% set variant = "--" + params.variant if params.variant and not params.hub %}
```
This is simpler and but you also then need to make my other variant comment

File: src/components/summary/_macro.njk [hunk 6]
```diff
@@ -1,6 +1,8 @@
...
+    {% set cardVariantwithMargin = "--card ons-u-mb-l" if params.variant == "card" and not params.hub else ""%}
```
Comment (rmccar) [inline src/components/summary/_macro.njk:5]: `cardVariantwithMargin` isnt being used

File: src/components/summary/example-summary-card-grouped.njk [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ]   
+            }
+        ]
+    })
+}}
```
Comment (rmccar) [inline src/components/summary/example-summary-card-grouped.njk:352]:
```suggestion
}}

```

File: src/components/summary/_macro.njk (hunk 6 above)
Comment (SriHV) [reply src/components/summary/_macro.njk:5]: It should be removed apologies for the confusion

File: src/components/summary/_macro.njk (hunk 4 above)
Comment (SriHV) [reply src/components/summary/_macro.njk:22]: Can we discuss this call tomorrow. I have few doubts regarding this

**Review by rmccar — COMMENTED** [review]
Some minor formatting changes in my comments and the border between the rows here needs updating to the light version

Screenshot 2024-02-08 at 14 31 00

**Comment by rmccar** [comment]
A border has been added to the bottom of the `example-summary-multiple` summary

Screenshot 2024-02-08 at 14 34 05

File: src/components/summary/_macro.njk [hunk 8]
```diff
@@ -1,34 +1,40 @@
 {% macro onsSummary(params) %}
     {% set className = "ons-summary" %}
     {% set titleSize = "2" %}
-
+    {% if params.variant and not params.hub%}
```
Comment (rmccar) [inline src/components/summary/_macro.njk:4]: What is the reason we don't want to offer this on hub summaries?

**Comment by rmccar** [comment]
There is a border added to summaries that have a link at the bottom that stretches right across the card

Screenshot 2024-02-08 at 14 37 02

**Comment by rmccar** [comment]
This looks a bit odd when there is an error, is it supposed to look like this?

Screenshot 2024-02-08 at 14 39 10

File: src/components/summary/example-summary-card-grouped.njk (hunk 7 above)
Comment (rmccar) [inline src/components/summary/example-summary-card-grouped.njk:352]: Some formatting changes here

```suggestion
{{
    onsSummary({
        "variant": "card",
        "summaries": [
            {
                "summaryTitle": "John Doe",
                "groups": [
                    {
                        "groupTitle": "Personal details",
                        "rows": [
                            {
                                "rowTitle": "Are you John Doe?",
                                "rowItems": [
                                    {
                                        "valueList": [
                                            {
                                                "text": "Yes I am"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    }
                                ]
                            },
                            {
                                "rowTitle": "What's your date of birth?",
                                "rowItems": [
                                    {
                                        "valueList": [
                                            {
                                                "text": "1 January 1981"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    }
                                ]
                            },
                            {
                                "rowTitle": "What is your sex?",
                                "rowItems": [
                                    {
                                        "valueList": [
                                            {
                                                "text": "Male"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    }
                                ]
                            }
                        ]
                    },
                    {
                        "groupTitle": "Identity and health",
                        "rows": [
                            {
                                "rowTitle": "What is your country of birth?",
                                "rowItems": [
                                    {
                                        "valueList": [
                                            {
                                                "text": "England"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    }
                                ]
                            },
                            {
                                "rowTitle": "What passports do you hold?",
                                "rowItems": [
                                    {
                                        "valueList": [
                                            {
                                                "text": "United Kingdom"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    }
                                ]
                            }
                        ]
                    },
                    {
                        "groupTitle": "Qualifications",
                        "rows": [
                            {
                                "rowTitle": "Have you completed an apprenticeship?",
                                "rowItems": [
                                    {
                                        "valueList": [
                                            {
                                                "text": "Yes"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    }
                                ]
                            },
                            {
                                "rowTitle": "Have you achieved a GCSE or equivalent qualification?",
                                "rowItems": [
                                    {
                                        "valueList": [
                                            {
                                                "text": "5 GCSEs grades A* to C or 9 to 4"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    }
                                ]
                            }
                        ]
                    },
                    {
                        "groupTitle": "Employment history",
                        "rows": [
                            {
                                "rowItems": [
                                    {
                                        "rowTitle": "Name of UK company",
                                        "valueList": [
                                            {
                                                "text": "Company A"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            },
                                            {
                                                "text": "Remove",
                                                "visuallyHiddenText": "Remove company",
                                                "url": "#0"
                                            }
                                        ]
                                    },
                                    {
                                        "rowTitle": "Head office location",
                                        "valueList": [
                                            {
                                                "text": "Cardiff"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    },
                                    {
                                        "rowTitle": "Is this UK company your current employer?",
                                        "valueList": [
                                            {
                                                "text": "No"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    }
                                ]
                            },
                            {
                                "rowItems": [
                                    {
                                        "rowTitle": "Name of UK company",
                                        "valueList": [
                                            {
                                                "text": "Company A"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            },
                                            {
                                                "text": "Remove",
                                                "visuallyHiddenText": "Remove company",
                                                "url": "#0"
                                            }
                                        ]
                                    },
                                    {
                                        "rowTitle": "Head office location",
                                        "valueList": [
                                            {
                                                "text": "Newport"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    },
                                    {
                                        "rowTitle": "Is this UK company your current employer?",
                                        "valueList": [
                                            {
                                                "text": "Yes"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    }
                                ]
                            }
                        ],
                        "summaryLink": {
                            "text": "Add a UK company or branch",
                            "url": "#0",
                            "attributes": {
                                "data-qa": "add-item-link"
                            }
                        }
                    },
                    {
                        "groupTitle": "Spending",
                        "rows": [
                            {
                                "rowTitle": "What are your monthly household expenses?",
                                "rowItems": [
                                    {
                                        "rowTitle": "Food",
                                        "valueList": [
                                            {
                                                "text": "£50.00"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    },
                                    {
                                        "rowTitle": "Utilities",
                                        "valueList": [
                                            {
                                                "text": "£65.00"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    },
                                    {
                                        "rowTitle": "Transport",
                                        "valueList": [
                                            {
                                                "text": "£70.00"
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    },
                                    {
                                        "rowTitle": "Other",
                                        "valueList": [
                                            {
                                                "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
                                            }
                                        ],
                                        "actions": [
                                            {
                                                "text": "Change",
                                                "visuallyHiddenText": "Change answer",
                                                "url": "#0"
                                            }
                                        ]
                                    }
                                ]
                            }
                        ]
                    }
                ]
            }
        ]
    })
}}
```

**Comment by SriHV** [comment]
> A border has been added to the bottom of the `example-summary-multiple` summary
> 
> Screenshot 2024-02-08 at 14 34 05



> There is a border added to summaries that have a link at the bottom that stretches right across the card
> 
> Screenshot 2024-02-08 at 14 37 02



> A border has been added to the bottom of the `example-summary-multiple` summary
> 
> Screenshot 2024-02-08 at 14 34 05

This is present in the main branch code also, should I change it?

**Comment by rmccar** [comment]
> > A border has been added to the bottom of the `example-summary-multiple` summary
> > Screenshot 2024-02-08 at 14 34 05
> 
> > There is a border added to summaries that have a link at the bottom that stretches right across the card
> > Screenshot 2024-02-08 at 14 37 02
> 
> > A border has been added to the bottom of the `example-summary-multiple` summary
> > Screenshot 2024-02-08 at 14 34 05
> 
> This is present in the main branch code also, should I change it?

Yeah ok forget the multiple one but the other one has the border but it doesn't look so odd because it doesn't stretch right across when all the other borders have gaps either side

**Review by rmccar — COMMENTED** [review]
Couple of things and also I think the VR snapshots need updating

**Comment by SriHV** [comment]
> Household summary with rows looks good but when there are no rows it doesn't look right anymore
> 
> Screenshot 2024-02-13 at 13 09 22

This problem exists in main branch as well when there are no rows, but will look into this!

**Comment by rmccar** [comment]
> > Household summary with rows looks good but when there are no rows it doesn't look right anymore
> > Screenshot 2024-02-13 at 13 09 22
> 
> This problem exists in main branch as well when there are no rows, but will look into this!

I don't think it is an issue in main, this is how it is meant to look https://service-manual.ons.gov.uk/design-system/components/summary/example-summary-household-no-rows I think the card version just needs some padding and the removal of the border

**Review by rmccar — COMMENTED** [review]
Other than these few things looks good

**Review by rmccar — APPROVED** [review]

File: src/components/summary/_summary.scss [hunk 9]
```diff
@@ -208,4 +208,36 @@ $hub-row-spacing: 1.3rem;
...
     }
   }
+
+  &__group {
+    &--card {
+      border: 1px solid var(--ons-color-borders-light);
+      .ons-summary__items {
+        padding: 0 1.25rem;
+        .ons-summary__item:not(:last-child) {
+          border-bottom: 1px solid var(--ons-color-borders-light);
```
Comment (adi-unni) [inline src/components/summary/_summary.scss:218]: Just wondering what IxD thinks of using light grey here to separate sections. The rest of the summary component uses the dark grey for borders

File: src/components/summary/example-summary-card-grouped.njk [hunk 10]
```diff
@@ -0,0 +1,352 @@
...
+                                                "visuallyHiddenText": "Change answer",
+                                                "url": "#0"
+                                            }
+                                        ]
+                                    },
+                                    {
+                                        "rowTitle": "Other",
+                                        "valueList": [
+                                            {
+                                                "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
```
Comment (adi-unni) [inline src/components/summary/example-summary-card-grouped.njk:333]: Need to make a note that you need some content design here

**Review by adi-unni — APPROVED** [review]
Great work on this! Remember to create a new card to add documentation around the new variant. Link this PR, and the new example you created to be used for the documentation.

**Pull request merged**